File: python/xinput_list.py
import subprocess

def list_xinput_devices():
    cmd = "xinput list"
    process = subprocess.Popen(cmd.split(), stdout=subprocess.PIPE)
    output, error = process.communicate()
    return [device for device in output.decode().split('\n') if "joystick" in device.lower()]

def print_xinput_devices():
    devices = list_xinput_devices()
    if not devices:
        print("No joystick devices found")
    else:
        print("List of joystick devices:")
        for device in devices:
            print(device)

if __name__ == "__main__":
    print_xinput_devices()

File: ruby/xinput_list.rb
def list_devices
    `xinput list | grep -i joystick`
  end

  def parse_devices(device_list)
    devices = []
    device_list.split("\n").each do |line|
      if line =~ /id=(\d+)/
        id = $1
        if line =~ /(.+)$/i
          name = $1.strip
          devices << {id: id, name: name}
        end
      end
    end
    devices
  end


  def main
    device_list = list_devices
    devices = parse_devices(device_list)
    devices.each do |device|
      puts "Joystick device '#{device[:name]}' (ID: #{device[:id]})"
    end
  end